drivers/sdl/SDL_build_config_private.h

```diff
@@ -0,0 +1,117 @@
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#define SDL_build_config_h_
+
+#include <SDL3/SDL_platform_defines.h>
+
+#define HAVE_STDARG_H 1
+#define HAVE_STDDEF_H 1
+
+// Here we disable SDL subsystems that are not going to be used
+#define SDL_CPUINFO_DISABLED 1
+#define SDL_AUDIO_DISABLED 1
+#define SDL_PROCESS_DUMMY 1
+#define SDL_LOADSO_DUMMY 1
+#define SDL_VIDEO_DISABLED 1
+#define SDL_CAMERA_DISABLED 1
+#define SDL_DIALOG_DISABLED 1
+#define SDL_FILESYSTEM_DUMMY 1
+#define SDL_FSOPS_DUMMY 1
+#define SDL_SENSOR_DISABLED 1
+#define SDL_GPU_DISABLED 1
+#define SDL_RENDER_DISABLED 1
+#define SDL_POWER_DISABLED 1
+#define SDL_LEAN_AND_MEAN 1
+
+// Windows defines
+#if defined(SDL_PLATFORM_WINDOWS)
+
+#define SDL_PLATFORM_PRIVATE_NAME "Windows"
+#define HAVE_LIBC 1
+#define HAVE_DINPUT_H 1
+#define HAVE_XINPUT_H 1
+#if defined(_WIN32_MAXVER) && _WIN32_MAXVER >= 0x0A00 /* Windows 10 SDK */
+#define HAVE_WINDOWS_GAMING_INPUT_H 1
+#define SDL_JOYSTICK_WGI 1
+#endif
+#define SDL_JOYSTICK_DINPUT 1
+#define SDL_JOYSTICK_HIDAPI 1
+#define SDL_JOYSTICK_RAWINPUT 1
+#define SDL_JOYSTICK_XINPUT 1
+#define SDL_HAPTIC_DINPUT 1
+#define SDL_THREAD_GENERIC_COND_SUFFIX 1
+#define SDL_THREAD_GENERIC_RWLOCK_SUFFIX 1
+#define SDL_THREAD_WINDOWS 1
+#define SDL_TIMER_WINDOWS 1
+
+// Linux defines
+#elif defined(SDL_PLATFORM_LINUX)
+
+#define SDL_PLATFORM_PRIVATE_NAME "Linux"
+#define SDL_PLATFORM_UNIX 1
+#define HAVE_STDIO_H 1
+#define HAVE_LINUX_INPUT_H 1
+#define HAVE_DBUS_DBUS_H 1
+#define HAVE_LIBUDEV_H 1
```

Comment:
It cannot be defined directly, but needs to be defined according to the detected conditions, because this library is optional.

https://github.com/godotengine/godot/blob/51b0379e5502402d44818a4b6f301f544a5754f3/platform/linuxbsd/detect.py#L383-L393

It seems possible to use `env["udev"]` as a conditional.

Reply:
See how it was done in `joypad_linux.cpp` before this PR:
```cpp
#ifdef UDEV_ENABLED
#ifdef SOWRAP_ENABLED
#include "libudev-so_wrap.h"
#else
#include <libudev.h>
#endif
#endif
```

`libudev-so_wrap.h` is a Godot wrapper that lets us dynamically load libudev.

Something similar is likely also needed for `HAVE_DBUS_DBUS_H`:
```cpp
#ifdef DBUS_ENABLED
#ifdef SOWRAP_ENABLED
#include "dbus-so_wrap.h"
#else
#include <dbus/dbus.h>
#endif
#endif
````

Reply:
On the other hand, the `SOWRAP_ENABLED` logic implies:
- We try to dlopen the library, and if it exists, we use it
- If it doesn't exist, we disable the feature

So it's a runtime check, while SDL here seems to need it to be defined at compile time.

So that might be a bit tricky and require SDL modifications to attempt dlopening like we do here:

https://github.com/godotengine/godot/blob/03bd8ba9c23d31f8f658f97093fd3f2e4a0f9031/platform/linuxbsd/joypad_linux.cpp#L94-L114

Reply:
Unless you see how to do this easily, I'd suggest reverting to the previous state for now, as this was close to being ready to merge in time for the feature freeze (2 days ago, but it's been postponed a couple of days as we're finalizing a few big changes, like this one).

We can work on adding libudev (and dbus?) dynamic loading in a follow-up.